provisioner/provision.go

```diff
@@ -0,0 +1,50 @@
+package provisioner
+
+import (
+	"fmt"
+
+	"github.com/resin-os/resin-provisioner/defaults"
+	"github.com/resin-os/resin-provisioner/util"
+)
+
+func Provision(appId, apiKey, userId, userName, configPath, domain string, dryRun bool) error {
+	deviceType, err := util.ScanDeviceTypeSlug(defaults.OSRELEASE_PATH)
+	if err != nil {
+		return err
+	}
+
+	config := Config{
+		AppId:            appId,
+		ApiKey:           apiKey,
+		UserId:           userId,
+		UserName:         userName,
+		DeviceType:       deviceType,
+		ApiEndpoint:      fmt.Sprintf("https://api.%s", domain),
+		RegistryEndpoint: fmt.Sprintf("registry.%s", domain),
+		VpnEndpoint:      fmt.Sprintf("vpn.%s", domain),
+	}
```

Comment:
All it does is write the same fields to the `config.json` as are present in a new, never booted, image from the dashboard and lets the supervisor do the rest.